torch/csrc/jit/serialization/python_print.cpp

```diff
@@ -1394,6 +1412,19 @@ struct PythonPrintImpl {
       for (auto& method : classType->methods()) {
         printFunction(*method);
       }
+      std::set<std::string> already_printed;
+      for (auto& hook : classType->getForwardHooks()) {
+        if (already_printed.count(hook->name()) == 0) {
+          already_printed.insert(hook->name());
+          printFunction(*hook);
+        }
+      }
+      for (auto& pre_hook : classType->getForwardPreHooks()) {
+        if (already_printed.count(pre_hook->name()) == 0) {
+          already_printed.insert(pre_hook->name());
+          printFunction(*pre_hook);
+        }
+      }
```

Comment:
It's probably a corner case but this might lead to bloating of serialized file size if the same hook is registered on multiple modules that are saved into the same archive.

Reply:
Is the bloat that much of a concern? The hooks should take up the same amount of space as other methods on a module

Reply:
Sure, but won't they be repeated in the body of the definition of every module that uses them?

I guess we can emit them inside the module definition for now, but let's keep this in mind in case the file size of a serialized archive of a module with hooks attached becomes problematic in the future. We should probably file an issue so that we don't lose track of it.